src/App.tsx

```diff
@@ -15,48 +15,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+// Use this function in the render method to prepare goods
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  // To avoid the original array mutation
+  const visibleGoods = [...goods];
+
+  if (sortType === SortType.ALPHABET) {
+    visibleGoods.sort((a, b) => a.localeCompare(b));
+  }
+
+  if (sortType === SortType.LENGTH) {
+    visibleGoods.sort((a, b) => a.length - b.length);
+  }
+  // Sort and reverse goods if needed
+  // eslint-disable-next-line no-console
+
+  return isReversed ? visibleGoods.reverse() : visibleGoods;
+}
+
+// DON'T save goods to the state
+// type State = {
+//   isReversed: boolean,
+//   sortType: SortType,
+// };
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [goods, setGoods] = useState(goodsFromServer);
+
+  useEffect(() => {
+    setGoods(getReorderedGoods(goods, { sortType, isReversed }));
```

Comment:
use callback to get current version of state
```suggestion
    setGoods(prevGoods => getReorderedGoods(prevGoods, { sortType, isReversed }));
```